tgui/packages/tgui/interfaces/ChemDispenser.js

```diff
@@ -0,0 +1,439 @@
+                "No " + beakerName + " Inserted"
+              ) || currentBeakerName + " Full"}
+            </Box>
+          </Modal>
+        )}
+        {dispensableReagents.map((reagent, reagentIndex) => (
+          <Button
+            key={reagentIndex}
+            className="chem-dispenser__dispense-buttons"
+            position="relative"
+            backgroundColor={""}
+            align="left"
+            width="129.5px"
+            onMouseEnter={() => setHoverOverId(reagent.id)}
+            onMouseLeave={() => setHoverOverId("")}
+            disabled={maximumBeakerVolume === beakerTotalVolume}
+            lineHeight={1.75}
+            onClick={() => act("dispense", {
+              amount: addAmount, reagentId: reagent.id,
+            })}>
+            <Icon
+              pt={1}
+              style={{
+                "text-shadow": "0 0 3px #000",
+              }}
+              color={"rgba(" + reagent.colorR + "," + reagent.colorG + ", " + reagent.colorB + ", 1)"}
+              name={iconToggle ? stateMap[reagent.state].icon : "circle"} />
+            <Tooltip.Overflow width="105px" content={reagent.name}
+              style={{
+                "line-height": "15px",
+              }}>
+              {reagent.name}
+            </Tooltip.Overflow>
+          </Button>
+        ))}
+      </Section>
+      <Box italic pt={0.5}> {"Reagent ID: " + hoverOverId}</Box>
+    </Section>
+  );
+};
+
+export const Beaker = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    beakerName,
+    maximumBeakerVolume,
+    currentBeakerName,
+    beakerTotalVolume,
+  } = data;
+
+  const [iconToggle] = useSharedState(context, 'iconToggle', false);
+  const [removeAmount, setRemoveAmount] = useLocalState(context, 'removeAmount', 20);
+  const removeReagentButtons = [removeAmount, 10, 5, 1];
+  const beakerContents = data.beakerContents || [];
+
+  return (
+    <Section
+      fontSize="12px"
+      title={
+        <Button
+          className="chem-dispenser__buttons"
+          icon="eject"
+          onClick={() => act("eject")}>
+          {!maximumBeakerVolume ? "Insert " + beakerName : "Eject " + currentBeakerName + " (" + beakerTotalVolume + "/" + maximumBeakerVolume + ")"}
+        </Button>
+      }
+      buttons={(
+        <Box align="left" as="span">
+          {"Remove Amount: "}
+          <NumberInput
+            width={4}
+            format={value => value + "u"}
+            value={removeAmount}
+            minValue={1}
+            maxValue={100}
+            onDrag={(e, value) => setRemoveAmount(value)} />
+        </Box>
+      )}>
+      <Table.Row>
+        <Table.Cell bold collapsing textAlign="center" />
+        <Table.Cell collapsing />
+      </Table.Row>
+      <Box color="label">
+        {!beakerContents.length && (
+          "No Contents"
+        )}
+      </Box>
+      {beakerContents.map((reagent, indexContents) => (
+        <Table.Row key={indexContents}>
```

Comment:
As I think the reagent key is unique, could use an identifier on `reagent` here. Indexes are a last resort key.